### PR TITLE
Feat/st 1051: Fix for Maintain Case links

### PR DESCRIPTION
**Pull request by sjamihmcts**
### Change description ###

Fix for Maintain Case links. Updated Delete permission to roles

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ST-1051

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/access/DefaultAccess.java [hunk 1]
```diff
@@ -19,10 +19,10 @@ public class DefaultAccess implements HasAccessControl {
...
         SetMultimap<HasRole, Permission> grants = HashMultimap.create();
         grants.putAll(CREATOR, Permissions.CREATE_READ_UPDATE);
-        grants.putAll(ST_CIC_CASEWORKER, Permissions.CREATE_READ_UPDATE);
-        grants.putAll(ST_CIC_SENIOR_CASEWORKER, Permissions.CREATE_READ_UPDATE);
-        grants.putAll(ST_CIC_HEARING_CENTRE_ADMIN, Permissions.CREATE_READ_UPDATE);
-        grants.putAll(ST_CIC_HEARING_CENTRE_TEAM_LEADER, Permissions.CREATE_READ_UPDATE);
+        grants.putAll(ST_CIC_CASEWORKER, Permissions.CREATE_READ_UPDATE_DELETE);
+        grants.putAll(ST_CIC_SENIOR_CASEWORKER, Permissions.CREATE_READ_UPDATE_DELETE);
+        grants.putAll(ST_CIC_HEARING_CENTRE_ADMIN, Permissions.CREATE_READ_UPDATE_DELETE);
+        grants.putAll(ST_CIC_HEARING_CENTRE_TEAM_LEADER, Permissions.CREATE_READ_UPDATE_DELETE);
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/access/DefaultAccess.java:25]: @sjamihmcts - Can we have the respective test file updated as well?

**Review by rajaram-nagarajan — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/access/DefaultAccess.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/access/DefaultAccess.java:25]: Done

**Review by osheaf01 — APPROVED** [review]
LGTM

**Review by Shailesh-walke — APPROVED** [review]

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

**Pull request merged**